oxd-server/src/main/java/org/gluu/oxd/server/mapper/RegisterResponseMapper.java

```diff
@@ -0,0 +1,447 @@
+                BlockEncryptionAlgorithm.fromName(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ENC.toString())) != null) {
+            rpFromRegisterResponse.setRequestObjectEncryptionEnc(respClaims.get(REQUEST_OBJECT_ENCRYPTION_ENC.toString()));
+        }
+
+        rpFromRegisterResponse.setDefaultMaxAge(safeToNumber(respClaims.get(DEFAULT_MAX_AGE.toString())));
+
+        if (!Strings.isNullOrEmpty(respClaims.get(REQUIRE_AUTH_TIME.toString()))) {
+            rpFromRegisterResponse.setRequireAuthTime(Boolean.valueOf(respClaims.get(REQUIRE_AUTH_TIME.toString())));
+        }
+
+        rpFromRegisterResponse.setInitiateLoginUri(respClaims.get(INITIATE_LOGIN_URI.toString()));
+        rpFromRegisterResponse.setAuthorizedOrigins(convertStringToList(respClaims.get(AUTHORIZED_ORIGINS.toString())));
+        rpFromRegisterResponse.setAccessTokenLifetime(safeToNumber(respClaims.get(ACCESS_TOKEN_LIFETIME.toString())));
+        rpFromRegisterResponse.setSoftwareId(respClaims.get(SOFTWARE_ID.toString()));
+        rpFromRegisterResponse.setSoftwareVersion(respClaims.get(SOFTWARE_VERSION.toString()));
+        rpFromRegisterResponse.setSoftwareStatement(respClaims.get(SOFTWARE_STATEMENT.toString()));
+        rpFromRegisterResponse.setClientJwksUri(respClaims.get(JWKS_URI.toString()));
+        rpFromRegisterResponse.setClaimsRedirectUri(convertStringToList(respClaims.get(CLAIMS_REDIRECT_URIS.toString())));
+
+        return rpFromRegisterResponse;
+    }
+
+    private static List<String> convertStringToList(String input) {
+        if (input == null)
+            return null;
+
+        String commaSeparatedString = input.replaceAll("\\[", "").replaceAll("\\]", "").replace("\"", "");
+        return Lists.newArrayList(commaSeparatedString.split(","));
+    }
+
+    private static List<String> convertSpaceSeparatedStringToList(String input) {
+        if (input == null)
+            return null;
+
+        return Lists.newArrayList(input.split(" "));
+    }
+
+    public static Integer safeToNumber(String num) {
+        if (num == null)
+            return null;
+        try {
+            return Integer.valueOf(num);
+        } catch (NumberFormatException ex) {
+            return null;
+        }
+    }
+
+    public static boolean compareParams(List<String> oxdRp, List<String> oxAuthRp) {
+        if (oxdRp == null || oxdRp.isEmpty()) {
+            return oxAuthRp == null || oxdRp.isEmpty();
+        } else if (oxAuthRp != null) {
+            return new HashSet<>(oxdRp).equals(new HashSet<>(oxAuthRp));
+        }
+        return false;
+    }
+
+    public static boolean compareJsonNodeParams(String oxdParam, String oxAuthparam) throws IOException {
+        return oxdParam == null ? oxAuthparam == null : Jackson2.createJsonMapperWithoutEmptyAttributes().readTree(oxdParam).equals(Jackson2.createJsonMapperWithoutEmptyAttributes().readTree(oxAuthparam));
+    }
+
+    public static boolean compareParams(Boolean oxdParam, Boolean oxAuthparam) {
```

Comment:
Such method already exist `java.util.Objects#equals()`. You don't need copy/paste for Boolean, Integer, etc. All these objects already have equals and nullity check is performed by `java.util.Objects#equals()`.
```
Boolean b1 = false;
Boolean b2 = null
boolean areSame = Objects.equals(b1, b2);
```

Reply:
Done